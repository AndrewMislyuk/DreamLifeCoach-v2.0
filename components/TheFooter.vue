<template>
  <div class="footer">
    <div class="footer__socials">
      <a
        v-for="(link, index) in links"
        :key="index"
        :href="link.href"
        class="footer__socials-link"
        target="_blank"
      >
        <div :class="`${link.icon} --icon`" />
      </a>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {
      links: [
        {
          icon: 'fab fa-instagram',
          href: 'https://www.instagram.com/liudmilanegoda/',
        },

        {
          icon: 'fab fa-facebook-square',
          href: 'https://www.facebook.com/LiudmilaNegoda',
        },

        {
          icon: 'fab fa-telegram',
          href: 'https://web.telegram.org/',
        },
      ],
    }
  },
}
</script>

<style lang="scss" scoped>
@import 'sass-rem/_rem.scss';

.footer {
  background-color: #2e2e2e;
  color: #fff;
  padding: rem(50px 0);

  &__socials {
    max-width: rem(300px);
    margin: 0 auto;
    display: flex;
    justify-content: space-between;
    align-items: center;

    &-link {
      color: #fff;
      font-size: rem(25px);
      width: rem(50px);
      height: rem(50px);
      border-radius: 50%;
      border: 1px solid #fff;
      cursor: pointer;
      position: relative;
      transition: all 0.2s linear;

      &:hover {
        border-color: #4caf50;
        color: #4caf50;
      }

      .--icon {
        position: absolute;
        top: 50%;
        left: 50%;
        margin-top: rem(-14px);
        margin-left: rem(-11px);
      }
    }
  }
}
</style>
