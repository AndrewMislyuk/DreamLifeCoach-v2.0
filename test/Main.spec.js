import { shallowMount } from '@vue/test-utils'
import TheFooter from '@/components/TheFooter.vue'

const factory = () => {
  return shallowMount(TheFooter, {})
}

describe('TheFooter', () => {
//   test('is a Vue instance', () => {
//     const wrapper = mount(TheFooter)
//     expect(wrapper.isVueInstance()).toBeTruthy()
//   })

  test('renders properly', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
