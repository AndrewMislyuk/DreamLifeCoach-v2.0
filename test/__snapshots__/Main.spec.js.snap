// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheFooter renders properly 1`] = `"<div class=\\"footer\\"></div>"`;
